core/system/inc/svc_gw.h

```diff
@@ -55,12 +55,14 @@ static inline uint32_t svc_gw_get_dst_fn(TSecGw *svc_pc)
 
 static inline uint8_t svc_gw_get_dst_id(TSecGw *svc_pc)
 {
-    uint32_t box_id = svc_pc->cfg_ptr - __uvisor_config.cfgtbl_ptr_start;
+    uint8_t box_id;
 
-    if(box_id <= 0 || box_id >= g_vmpu_box_count)
```

Comment:
This condition seems different from the `vmpu_is_box_id_valid` condition. Consider fixing it in a separate commit before refactoring to use `vmpu_is_box_id_valid`. This makes it obvious that the condition is changing.

Specifically, this wouldn't allow a box_id of 0, but `vmpu_is_box_id_valid` does allow a box_id of 0. This might be a bug deserving a fix in its own commit, or we might be introducing a bug by starting to use `vmpu_is_box_id_valid` here.